mlflow/store/sqlalchemy_store.py

```diff
@@ -347,6 +391,7 @@ def delete_run(self, run_id):
             self._save_to_db(objs=run, session=session)
 
     def log_metric(self, run_id, metric):
+        _validate_metric(metric.key, metric.value, metric.timestamp, metric.step)
```

Comment:
Validate the metric at the store level to get a MySQL test (`test_log_null_metric`) to pass - MySQL otherwise inserts a metric with `metric.value = None` as a metric with value 0. In the future we might consider validating metric entity fields at construction time so we don't need to duplicate calls to e.g. `_validate_metric` across stores